### PR TITLE
merge for error fixing, logging system, restarting system, HR parsing

### PR DESCRIPTION
**Pull request by keomabrun**

### CODE REVIEW

File: logging.conf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Note Expects 'logDir' passed in with location for file output.
```
Comment (twatteyne) [inline logging.conf]: remove

File: logging.conf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+#============================ formatters ======================================
+
+[formatters]
+keys=std,console
+
+[formatter_std]
```
Comment (twatteyne) [inline logging.conf]: rename `_std` to `_file`?

File: logging.conf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+class=StreamHandler
+args=()
+formatter=console
+
+#============================ loggers =========================================
+
+[loggers]
+keys=root,solmanager
+
+[logger_root]
```
Comment (twatteyne) [inline logging.conf]: please add comment about what this is. I believe `root` refers to the default one?

File: solmanager.config.sample [hunk 4]
```diff
@@ -1,24 +1,18 @@
...
-
-# Application
-filecommitdelay = 60
-sendperiodminutes = 1
-fileperiodminutes = 1
-
-# Security
-token = DEFAULT_SOLMANAGERTOKEN
+[application]
+logfile             = solmanager.log
```
Comment (twatteyne) [inline solmanager.config.sample]: remove?

File: solmanager.config.sample [hunk 5]
```diff
@@ -1,24 +1,18 @@
...
+backupfile          = solmanager.backup
+serialport          = /dev/ttyUSB3
+tcpport             = 8080
+filecommitdelay     = 60
```
Comment (twatteyne) [inline solmanager.config.sample]: difference between `filecommitdelay` and `fileperiodminutes`?
Please add comments to every line

File: solmanager.config.sample [hunk 6]
```diff
@@ -1,24 +1,18 @@
...
+fileperiodminutes   = 1
 
-# Files
-crashlogfile = solmanager.crashlog
-backupfile = solmanager.backup
-
-[server]
-# Connection
-host = localhost:8081
+[solmanager]
```
Comment (twatteyne) [inline solmanager.config.sample]: group all configs?

File: solmanager.config.sample [hunk 7]
```diff
@@ -1,24 +1,18 @@
...
+token               = SOLMANAGER_TOKEN
+cert                = solmanager.cert
+privkey             = solmanager.ppk
 
-# Security
-token = DEFAULT_SOLSERVERTOKEN
-certfile = solserver.cert
+[solserver]
```
Comment (twatteyne) [inline solmanager.config.sample]: old config of solserver. remove?

File: solmanager.py [hunk 8]
```diff
@@ -15,26 +15,25 @@
 import time
 import threading
 import json
+import subprocess
```
Comment (twatteyne) [inline solmanager.py]: remove if not used

File: solmanager.py [hunk 9]
```diff
@@ -45,64 +44,68 @@
 import SolVersion
 import SolDefines
 
+#============================ logging =========================================
+
+logging.config.fileConfig('logging.conf')
+log = logging.getLogger('solserver')
```
Comment (twatteyne) [inline solmanager.py]: rename `solserver` to `solmanager`

File: logging.conf (hunk 1 above)
Comment (twatteyne) [inline logging.conf]: shouldn't you need a logging configuration for `sol` logger as well?

File: solmanager.py [hunk 10]
```diff
@@ -45,64 +44,68 @@
...
-DEFAULT_LOGFILE                        = 'solmanager.sol'
-DEFAULT_SOLSERVER                      = 'localhost:8081'
-DEFAULT_SOLSERVERTOKEN                 = 'DEFAULT_SOLSERVERTOKEN'
-DEFAULT_SOLMANAGERTOKEN                = 'DEFAULT_SOLMANAGERTOKEN'
-DEFAULT_SOLMANAGERPRIVKEY              = 'solmanager.ppk'
-DEFAULT_SOLMANAGERCERT                 = 'solmanager.cert'
-DEFAULT_SOLSERVERCERT                  = 'solserver.cert'
-DEFAULT_SENDPERIODMINUTES              = 1
-DEFAULT_FILEPERIODMINUTES              = 1
+LOGFILE                                 = ''
```
Comment (twatteyne) [inline solmanager.py]: in general, just use the config from the config file directly, don't populate a global var

File: solmanager.py [hunk 11]
```diff
@@ -474,81 +481,84 @@ def runHardware(self):
...
-                    #    130, 34, 8, 8, 0, 2, 215, 2, 0, 3, 221, 2, 0, 5, 200, 2, 0, 6, 211, 2, 0, 8, 214, 2, 0, 9, 220, 2, 0, 10, 199, 2, 0, 11, 214, 2 <- HR_ID_DISCOVERED
-                    #]
-                    dust_notifs = [
-                        IpMgrConnectorSerial.IpMgrConnectorSerial.Tuple_notifHealthReport(
-                            macAddress = dust_notif.macAddress,
-                            payload    = dust_notif.payload[:dust_notif.payload[1]+2],
-                        ),
+                hr_exists       = True
+                dust_notifs     = []
+                hr_currptr  = 0
```
Comment (twatteyne) [inline solmanager.py]: indent `=`

File: solmanager.py [hunk 12]
```diff
@@ -1227,15 +1241,27 @@ def _exec_cmd(self,cmd):
...
         self.sendThread      = SendThread()
-        self.jsonThread      = JsonThread(tcpport,self.dustThread)
-    
+        self.jsonThread      = JsonThread(TCPPORT,self.dustThread)
+        log.debug("All threads stated")
+
+    def restart(self):
+        log.debug("Restarting threads")
+        self.close()
+        time.sleep(3)
```
Comment (twatteyne) [inline solmanager.py]: add comment to say this isn't a race condition

Comment (twatteyne) [inline solmanager.py]: maybe loop through threads and verify they are all inactive before continuing

File: solmanager.py [hunk 13]
```diff
@@ -1227,15 +1241,27 @@ def _exec_cmd(self,cmd):
...
+        self.dustThread      = DustThread(SERIALPORT,simulation=False)
```
Comment (twatteyne) [inline solmanager.py]: don't copy-paste this and  the `__init__()` code, rather implement a `start()` method

File: solmanager.py [hunk 14]
```diff
@@ -1314,55 +1335,64 @@ def main(serialport,tcpport):
...
+    SOLMANAGER_TOKEN    = cf_parser.get('solmanager','token')
+    SOLMANAGER_CERT     = cf_parser.get('solmanager','cert')
+    SOLMANAGER_PRIVKEY  = cf_parser.get('solmanager','privkey')
+
+    # solserver config
+    SOLSERVER_HOST      = cf_parser.get('solserver','host')
+    SOLSERVER_TOKEN     = cf_parser.get('solserver','token')
+    SOLSERVER_CERT      = cf_parser.get('solserver','certfile')
+
+    log.debug("Configuration:\n" +\
```
Comment (twatteyne) [inline solmanager.py]: - use `format()`
- use a loop?

File: solmanager.config.sample (hunk 4 above)
Comment (keomabrun) [inline solmanager.config.sample]: Comments on log fixes:
https://github.com/realms-team/solmanager/commit/1791dc136917518742354c3df12f9411052c74a3

**Pull request merged**